### PR TITLE
pretraining config fix

### PR DESCRIPTION
**Pull request by mshuaibii**
Loading a pretrained model previously requires the exact config specified (including data) to be available to use. This PR makes it such that:
- configs are saved during training in a `config.pt` file that contains all the hyperparameters necessary to reconstruct the model, normalizers, etc;
- A pretrained model may be loaded with a blank config and all the necessary information is extracted from the pretrained checkpoint:
```
trainer = AtomsTrainer()
trainer.load_pretrained(checkpoint_path)
trainer.predict(images)
```
- Include tests to verify that prediction results are consistent across trainers that train/predict in one sitting, trainers that load the previous checkpoint (with a config available identical to that used during training), and trainers that have no config and just load the same checkpoint
- Clean up some fingerprinting/preprocessing structures

Various portions of this impact some of you, can you guys review and let me know. Thanks!

Few things I'd like to clean up with how this is organized, but this gets the job done for now.

@ray38 @ruiqic @EricMusa

### CODE REVIEW

**Review by ruiqic — APPROVED** [review]
lgtm

**Review by EricMusa — APPROVED** [review]
Good to go, good changes! (apologies if this was waiting on me)
